src/main/perl/Process.pm

```diff
@@ -165,16 +304,26 @@ sub _LC_Process
     if ($self->noAction()) {
         $self->debug(1, "LC_Process in noaction mode for $function");
         $? = 0;
-        return $noaction_value;
+        $res = $noaction_value;
     } else {
-        my $funcref = $LC_PROCESS_DISPATCH{$function};
-        if (defined($funcref)) {
-            return $funcref->(@$args);
-        } else {
-            $self->error("Unsupported LC::Process function $function");
-            return;
+        # The original GID (as list of groups)
+        my $orig_user_group = [$UID, "$GID"];
+
+        if ($self->_set_eff_user_group()) {
```

Comment:
I think my main reservation about this PR is that setting the user/group is going to be a rare code path, the common case will remain that user/group are unset and things run as root. So that means for someone try to understand the code they now have to read  three extra levels of function depth to confirm that it does nothing.

So while it's redundant to do so, I think from a code maintenance point of view it would be better to gate the call to _set_eff_user_group() on user/group being set.

Reply:
i understand.

so you mean something like 
```perl
my $current_user_group = (!defined $self->{user}) or (!defined $self->{group});
if ($current_user_group or $self->_set_eff_user_group()) {
 ...
}
```
and an extra conditional `if ! $current_user_group` for the restore part?